File: scripts/check-v2-estudo-content.mjs
import fs from "fs";
import path from "path";

function strip(t) {
  return t.normalize("NFD").replace(/\p{M}/gu, "");
}

const dir = "v2/parts";
const files = fs.readdirSync(dir).filter((f) => f.endsWith("-content.json")).sort();
console.log("file|part_id|title|notes|bullets|chars|oque|quando|ex|arm|lig|stub");
for (const f of files) {
  const s = JSON.parse(fs.readFileSync(path.join(dir, f), "utf8"));
  const notes = s.study_notes || [];
  const bullets = notes.reduce((n, x) => n + (x.bullets || []).length, 0);
  const chars = notes.reduce(
    (n, x) => n + (x.bullets || []).reduce((m, b) => m + b.length, 0),
    0
  );
  const heads = notes.map((h) => strip((h.heading || "").toLowerCase()));
  const hasOque = heads.some((h) => h.includes("o que"));
  const hasQuando = heads.some((h) => h.includes("quando"));
  const hasEx = heads.some(
    (h) => h.includes("exemplo") || h.includes("cenario") || h.includes("lab")
  );
  const hasArm = heads.some((h) => h.includes("armadilha"));
  const hasLig = heads.some((h) => h.includes("simulado") || h.includes("trilha"));
  const blob = JSON.stringify(notes);
  const stub =
    chars < 400 ||
    blob.includes("TODO") ||
    blob.toLowerCase().includes("em breve") ||
    blob.toLowerCase().includes("lorem");
  console.log(
    [
      f,
      s.part_id,
      String(s.title || "").slice(0, 42),
      notes.length,
      bullets,
      chars,
      hasOque,
      hasQuando,
      hasEx,
      hasArm,
      hasLig,
      stub,
    ].join("|")
  );
}

File: scripts/check-v2-estudo-content.py
"""Gate de qualidade do Estudo CCNA V2 (study_notes)."""
import json
import unicodedata


def strip(t: str) -> str:
    return "".join(
        c
        for c in unicodedata.normalize("NFD", t)
        if unicodedata.category(c) != "Mn"
    )


# IDs esperados pelo gen (arquivos em v2/parts)
import os

files = sorted(
    f
    for f in os.listdir("v2/parts")
    if f.endswith("-content.json")
)
print("file|part_id|title|notes|bullets|chars|oque|quando|ex|arm|lig|stub|ok_json")
for f in files:
    p = os.path.join("v2/parts", f)
    try:
        s = json.load(open(p, encoding="utf-8"))
        ok = True
    except Exception as e:
        print(f"{f}|INVALID|{e}")
        continue
    notes = s.get("study_notes") or []
    bullets = sum(len(x.get("bullets") or []) for x in notes)
    chars = sum(len(b) for x in notes for b in (x.get("bullets") or []))
    heads = [strip(h.get("heading", "").lower()) for h in notes]
    has_oque = any("o que" in h for h in heads)
    has_quando = any("quando" in h for h in heads)
    has_ex = any("exemplo" in h or "cenario" in h or "lab" in h for h in heads)
    has_arm = any("armadilha" in h for h in heads)
    has_lig = any("simulado" in h or "trilha" in h for h in heads)
    blob = str(notes)
    stub = (
        chars < 400
        or "TODO" in blob
        or "em breve" in blob.lower()
        or "lorem" in blob.lower()
        or "neste modulo voce" in strip(blob.lower())
    )
    print(
        f,
        s.get("part_id"),
        (s.get("title") or "")[:48],
        len(notes),
        bullets,
        chars,
        has_oque,
        has_quando,
        has_ex,
        has_arm,
        has_lig,
        stub,
        ok,
        sep="|",
    )
